src/bigquery/bigquery.py

```diff
@@ -0,0 +1,142 @@
+                INNER JOIN `bigquery-public-data.github_repos.files` AS files
+                ON files.repo_name = langs.repo_name
+                    WHERE (
+                        (
+                            REGEXP_CONTAINS(TO_JSON_STRING(language), r'(?i)java') AND
+                            files.path LIKE '%{m}'
+                        ) OR
+                        (
+                            REGEXP_CONTAINS(TO_JSON_STRING(language), r'(?i)python') AND
+                            files.path LIKE '%{p}'
+                        ) OR
+                        (
+                            files.path LIKE '%{n}'
+                        )
+                    )
+            ) AS L
+            ON con.id = L.id;
+        """.format(m=ECOSYSTEM_MANIFEST_MAP['maven'],
+                   p=ECOSYSTEM_MANIFEST_MAP['pypi'],
+                   n=ECOSYSTEM_MANIFEST_MAP['npm'])
+
+
+class BQDataProcessing(DataProcessing):
+    """Big query data fetching and processing class."""
+
+    def __init__(self):
+        """Initialize the BigQueryDataProcessing object."""
+        s3_client = AmazonS3(
+            region_name=AWS_SETTINGS.s3_region,
+            bucket_name=AWS_SETTINGS.s3_bucket_name,
+            aws_access_key_id=AWS_SETTINGS.s3_access_key_id,
+            aws_secret_access_key=AWS_SETTINGS.s3_secret_access_key,
+            local_dev=not AWS_SETTINGS.use_cloud_services
+        )
+        super().__init__(s3_client)
+        self.big_query_instance = Bigquery()
+        self.collectors = {}
+        self.filename = '{}/big-query-data/{}'.format(
+            AWS_SETTINGS.deployment_prefix, AWS_SETTINGS.s3_collated_filename)
+
+    def process(self, validate=False):
+        """Process Bigquery response data."""
+        for ecosystem in ECOSYSTEM_MANIFEST_MAP.keys():
+            self.collectors[ecosystem] = self._get_collector(ecosystem)
+
+        start = time.monotonic()
+        index = 0
+        logger.info("Running Bigquery synchronously")
+        self.big_query_instance.run_query_sync()
+        for object in self.big_query_instance.get_result():
+            index += 1
+
+            path = object.get('path', None)
+            content = object.get('content', None)
+
+            if not path or not content:
+                logger.warning("Either path %s or content %s is null", path, content)
+                continue
+
+            ecosystem = None
+            for _ecosystem, manifest in ECOSYSTEM_MANIFEST_MAP.items():
+                if path.endswith(manifest):
+                    ecosystem = _ecosystem
+
+            if not ecosystem:
+                logger.warning("Could not find ecosystem for given path %s", path)
+                continue
+
+            self.collectors[ecosystem].parse_and_collect(content, validate)
+
+        logger.info("Processed %d manifests in time: %f", index, time.monotonic() - start)
+        self._update_s3()
+
+    def _get_collector(self, ecosystem=str) -> BaseCollector:
```

Comment:
This is an incorrect type hinting.
```suggestion
    def _get_collector(self, ecosystem: str) -> BaseCollector:
```